spec/amber/cli/build_spec.cr

```diff
@@ -0,0 +1,40 @@
+{% if flag?(:run_build_tests) %}
+require "../../spec_helper"
+
+module Amber::CLI
+  begin
+    describe "building a generated app" do
+      ENV["AMBER_ENV"] = "test"
+      MainCommand.run ["new", TESTING_APP]
+      Dir.cd(TESTING_APP)
+      MainCommand.run ["generate", "scaffold", "Animal", "name:string"]
+      Amber::CLI::Spec.prepare_yaml(Dir.current)
+
+      build_result = `shards build`
+      db_result = `amber db drop create migrate`
```

Comment:
Will this line be a problem for running in Travis or any docker container?

Reply:
Maybe?, I think we are providing `amber` command, or you mean database issues?

Reply:
Any issue that you can think of:
1. Will it be using `amber` the released shard?
1. Do we need to build amber from source and `../../bin/amber [command]`?
1. Will the database be available for creating, migrating etc.?

Reply:
@marksiemers It will need to use the current version of amber being tested. Not current release. I'll approve this as soon as the tests pass.

Reply:
@elorest - I'm likely going to need some help with the CSRF stuff in the controller spec file (if that is indeed one of the issues). I'm familiar with how CSRF works and the concepts - just not how to apply them in the context of a Spec file.

Right now, the scaffold generator creates something like this: https://www.zerobin.net/?ec8d0fb58964e348#hMC3Nx4/zE5DzaA+KnWHtjPXOwsjXve1BiaaxU3twCU=

And when you run `crystal spec`, you get this:
https://www.zerobin.net/?0bad105752bbbd44#Y+UQ+nT4AU6mwPXlT0tPXDCIg1KeRZjV5LC8yDVBd4E=

@drujensen 
It appears there may be an issue with `amber-spec`.

Although, when the compiler issue is resolved, there may be a CSRF issue waiting at runtime.

Reply:
It looks like the namespacing for this constant has changed 
from: `Amber::Router::Params::URL_ENCODED_FORM`
to: `Amber::Router::ParamsParser::URL_ENCODED_FORM`

Does `amber-spec` need to rely on that constant? It is `"application/x-www-form-urlencoded"`

It is [part of the HTML standard](https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1), so it shouldn't change too soon, right?

Reply:
@marksiemers For this PR does anything change regarding csrf tests?

Reply:
After resolving the compiler error, there is a CSRF error at runtime, see details of running `crystal spec` here: https://www.zerobin.net/?3066168cc0d1f33d#s6F3u71vr7hOAeszBB1MNmEO3WqFANbJd8VKZfiJ1RA=

Reply:
@elorest - Nothing in this PR changes CSRF tests.

To clarify, the tests that are failing are in the generated `sample-app` with a scaffolded `Animal`. The failure in that test suite causes a failure in this test suite.

You can reproduce by doing the following:
```bash
amber new test_app && cd test_app
```
Then update `shard.yml` to have:
```yml
dependencies:
  amber:
    github: amberframework/amber
    branch: master

development_dependencies:
  amber_spec:
    github: marksiemers/amber-spec
    branch: remove-amber-constant
``` 
then update the `database.yml` to make sure the url has `postgres` instead of `root` (if needed)

then some more bash:
```bash
shards update && amber g scaffold Animal name:string && amber db drop create migrate && crystal spec
```